docker-build

```diff
@@ -64,12 +103,49 @@ push() {
 
 run() {
   local repo="$1"
-  build "$repo" "$(pull "$repo")" && push "$repo"
+  shift
+  build_with_cache "$repo" "$@" && push "$repo"
 }
 
+# Main -------------------------------------------------
+
 case "$1" in
+  help|-h|--help) usage ;;
+esac
+
+if [ -n "$CIRCLECI" ]; then
+  REPO="$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME"
+  BRANCH="$CIRCLE_BRANCH"
+  SHA="$CIRCLE_SHA1"
+else
+  >&2 echo "This script only works on CircleCI right now."
```

Comment:
Was this always the case? Are you sure we don't use it anywhere outside of circle?

Reply:
Sorry, the diff is misleading because I moved this section - this was present before my changes. That being said, the reason it doesn't work outside CircleCI is that it relies on the presence of `$CIRCLE_BRANCH` etc.